### PR TITLE
Properly preserve chi_table license as mandated by BSD-3-Clause

### PR DESCRIPTION
**Pull request by georgthegreat**
Amend reference to online hosted file with the full license quotation as mandated by the original license.

### CODE REVIEW

**Comment by mshabunin** [comment]
Maybe it would be better to follow SoftFloat scheme instead and install license file separately? @asmorkalov , @vpisarev , what do you think? Otherwise we would not comply with the second license clause regarding binary redistribution.
* https://github.com/opencv/opencv/blob/8a1b998b2ba3993919deafd137a33f55fce2c962/modules/core/CMakeLists.txt#L199
* https://github.com/opencv/opencv/blob/4.x/modules/core/3rdparty/SoftFloat/COPYING.txt

**Comment by georgthegreat** [comment]
As I understand, only parts of the original code are actually used by opencv.
There is no need to copy chitab1 factors.

Plus I am not familiar with opencv vendoring routine so someone else is needed to do the trick.

**Comment by mshabunin** [comment]
@georgthegreat , I've updated your PR: license is now in a separate file which will be installed to the `<prefix>/share/licenses/opencv4/MSCR-chi_table_LICENSE.txt`.

**Comment by georgthegreat** [comment]
Actually, having this license inlined was intentional.
We use automatic licenses scanning (i. e. scancode-toolkit) and having the license text inlined is a better option.

**Comment by mshabunin** [comment]
Perhaps it would be better to extract chi_table.h to a separate file then...

File: modules/features2d/3rdparty/MSCR/chi_table.h [hunk 1]
```diff
@@ -0,0 +1,134 @@
+/*
```
Comment (opencv-alalek) [inline modules/features2d/3rdparty/MSCR/chi_table.h:1]: > modules/features2d/3rdparty/MSCR/chi_table.h
> MSCR

Please avoid uppercase in filenames.

File: modules/features2d/3rdparty/MSCR/chi_table.h [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+** warranties of merchantability and fitness for a particular purpose are disclaimed.
+** In no event shall the Intel Corporation or contributors be liable for any direct,
+** indirect, incidental, special, exemplary, or consequential damages
+** (including, but not limited to, procurement of substitute goods or services;
+** loss of use, data, or profits; or business interruption) however caused
+** and on any theory of liability, whether in contract, strict liability,
+** or tort (including negligence or otherwise) arising in any way out of
+** the use of this software, even if advised of the possibility of such damage.
+**
+*/
```
Comment (opencv-alalek) [inline modules/features2d/3rdparty/MSCR/chi_table.h:32]: `Content origin: http://users.isy.liu.se/cvl/perfo/software/chi_table.h`

**Review by asmorkalov — APPROVED** [review]
:+1:

**Pull request merged**